machine-learning-image-classification-with-javascript-and-nyckel/index.md

```diff
@@ -0,0 +1,327 @@
+{
+    var reader = new FileReader();
+    reader.onload = function (d)
+    {
+        $('.upload-area').css("background-image", "url(" + d.target.result + ")");
+    }
+    reader.readAsDataURL(file);
+}
+
+function resizeAndUploadImage(file)
+{
+    // Ensure it's an image
+    if (file.type.match(/image.*/))
+    {
+        // Load the image
+        var reader = new FileReader();
+        reader.onload = function (readerEvent)
+        {
+            var image = new Image();
+            image.onload = function (imageEvent)
+            {
+                // Resize the image
+                var canvas = document.createElement('canvas'),
+                    max_size = 600,
+                    width = image.width,
+                    height = image.height;
+                if (width > height)
+                {
+                    if (width > max_size)
+                    {
+                        height *= max_size / width;
+                        width = max_size;
+                    }
+                } else
+                {
+                    if (height > max_size)
+                    {
+                        width *= max_size / height;
+                        height = max_size;
+                    }
+                }
+                canvas.width = width;
+                canvas.height = height;
+                canvas.getContext('2d').drawImage(image, 0, 0, width, height);
+                var dataUrl = canvas.toDataURL('image/jpeg');
+                var resizedImage = dataURLToBlob(dataUrl);
+
+                checkImageWithNyckel(resizedImage);
+            }
+            image.src = readerEvent.target.result;
+        }
+        reader.readAsDataURL(file);
+    }
+    else
+    {
+        alert("You must choose an image");
+        resetPage();
+    }
+}
+```
+The rectangular area for hosting the uploaded image is called a **canvas** in html. A **blob** on the other hand represents objects in form of raw data. Converting canvas to blob translates the images uploaded to your application in a way that the model can understand.
+
+The code below is a utility function that further extends the JavaScript file to convert a canvas to a blob.
+
+```JavaScript
+/* Utility function to convert a canvas to a BLOB */
+var dataURLToBlob = function (dataURL)
+{
+    var BASE64_MARKER = ';base64,';
+    if (dataURL.indexOf(BASE64_MARKER) == -1)
+    {
+        var parts = dataURL.split(',');
+        var contentType = parts[0].split(':')[1];
+        var raw = parts[1];
+
+        return new Blob([raw], { type: contentType });
+    }
+
+    var parts = dataURL.split(BASE64_MARKER);
+    var contentType = parts[0].split(':')[1];
+    var raw = window.atob(parts[1]);
+    var rawLength = raw.length;
+
+    var uInt8Array = new Uint8Array(rawLength);
+
+    for (var i = 0; i < rawLength; ++i)
+    {
+        uInt8Array[i] = raw.charCodeAt(i);
+    }
+
+    return new Blob([uInt8Array], { type: contentType });
+}
+
+```

```

Comment:
Where is the expanation?

Reply:
I have written an explanation about canvas and blob... and indicated that the code is to convert canvas to blob